### PR TITLE
audit: audit formulae with the same name

### PR DESCRIPTION
**Pull request by xu-cheng**
Address #40152

### CODE REVIEW

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -161,6 +164,42 @@ def audit_class
...
+        missing_taps.each { |tap| Homebrew.install_tap "homebrew", tap }
+      end
+      same_name_formulae = Formula.full_names.group_by { |n| n.split("/").last }.fetch(name)
+      if same_name_formulae.size > 1
+        problem "Formula name is conflicted with #{(same_name_formulae - [full_name]).join ", "}"
+      end
+    ensure
+      nostdoutstderr do
+        missing_taps.each { |tap| Homebrew.uninstall_tap "homebrew", tap }
+      end
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: We definitely don't need to install each of these taps when we can use `brew search` instead.

Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: Updated to use search.

**Comment by xu-cheng** [comment]
I wonder is there any way to suppress this audit rule in the event of formula migration?

**Comment by mikemcquaid** [comment]
@xu-cheng Hmm. I think it's probably just fine to ignore it in that case? I'm open to any other ideas, though. Approach makes sense to me :+1:

**Comment by mikemcquaid** [comment]
Oh, last thing would be checking this fails nicely if you don't have an internet connection.

**Comment by xu-cheng** [comment]
> Oh, last thing would be checking this fails nicely if you don't have an internet connection.

`Homebrew.search_tap` already handled this. Currently such error would be just ignored silently.

**Comment by mikemcquaid** [comment]
:+1:

**Pull request closed without merging**